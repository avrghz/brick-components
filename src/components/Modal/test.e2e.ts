import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'

// ! add test case for bkClose

describe('<bk-modal/>', () => {
    let page: E2EPage
    let component: E2EElement

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-modal open="true"></bk-modal>`)
        component = await page.find('bk-modal')
    })

    describe('Prop(open)', () => {
        it('should render modal when set to true', async () => {
            expect(component).toMatchSnapshot()
        })

        // fixme : open=false not working with test
        it('should not render modal when set to false', async () => {
            page = await newE2EPage()
            await page.setContent(`<bk-modal open="false"></bk-modal>`)
            component = await page.find('bk-modal')
            expect(component).toMatchSnapshot()
        })
    })

    describe('Prop(width)', () => {
        it(`should render with given width`, async () => {
            component.setProperty('width', 75)
            await page.waitForChanges()
            expect(component).toMatchSnapshot()
        })
    })

    describe('Prop(dismissible)', () => {
        describe('when set to false', () => {
            it(`should render not render close button`, async () => {
                component.setProperty('dismissible', false)
                await page.waitForChanges()
                expect(component).toMatchSnapshot()
            })
        })
    })
})
