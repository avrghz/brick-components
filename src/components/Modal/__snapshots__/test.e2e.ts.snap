// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-modal/> Prop(dismissible) when set to false should render not render close button 1`] = `
<bk-modal aria-hidden="" aria-modal="true" class="hydrated" open="" role="dialog" tabindex="0">
  <mock:shadow-root>
    <div class="bk-modal">
      <div class="bk-modal__content open" role="document">
        <focus-trap>
          <main class="bk-modal__body">
            <slot name="body"></slot>
          </main>
        </focus-trap>
      </div>
      <div class="bk-modal__backdrop open"></div>
    </div>
  </mock:shadow-root>
</bk-modal>
`;

exports[`<bk-modal/> Prop(open) should not render modal when set to false 1`] = `
<bk-modal class="hydrated" open="false">
  <mock:shadow-root></mock:shadow-root>
</bk-modal>
`;

exports[`<bk-modal/> Prop(open) should render modal when set to true 1`] = `
<bk-modal aria-hidden="" aria-modal="true" class="hydrated" open="" role="dialog" tabindex="0">
  <mock:shadow-root>
    <div class="bk-modal">
      <div class="bk-modal__content open" role="document">
        <focus-trap>
          <main class="bk-modal__body">
            <slot name="body"></slot>
          </main>
        </focus-trap>
      </div>
      <div class="bk-modal__backdrop open"></div>
    </div>
  </mock:shadow-root>
</bk-modal>
`;

exports[`<bk-modal/> Prop(width) should render with given width 1`] = `
<bk-modal aria-hidden="" aria-modal="true" class="hydrated" open="" role="dialog" tabindex="0">
  <mock:shadow-root>
    <div class="bk-modal">
      <div class="bk-modal__content open" role="document">
        <focus-trap>
          <main class="bk-modal__body">
            <slot name="body"></slot>
          </main>
        </focus-trap>
      </div>
      <div class="bk-modal__backdrop open"></div>
    </div>
  </mock:shadow-root>
</bk-modal>
`;
